Library/Homebrew/cask/lib/hbc/cli/search.rb

```diff
@@ -13,6 +13,22 @@ def self.extract_regexp(string)
         end
       end
 
+      def self.search_remote(*arguments)
+        query = arguments.join(" ")
+
+        matches = GitHub.search_code(user: 'caskroom', filename: query)
+        matches.map do |match|
+          dirname, filename = File.split(match["path"])
+          next unless dirname == "Casks"
+          user = match["repository"]["owner"]["login"]
+          repo = match["repository"]["name"]
+          tap = Tap.fetch(user, repo)
+          next if tap.installed?
+          basename = File.basename(filename, ".rb")
+          "#{tap.name}/#{basename}"
+        end.compact
```

Comment:
This logic is very similar and looks like pretty much all except the specific dirname could be shared.